lib/Template.js

```diff
@@ -0,0 +1,54 @@
+try {
+  module.exports = require('./es6/Template')
+} catch (ignored) {
+  // ES6 code failed to load.
+  //
+  // This happens in Node runtimes with versions < 6.
+  // Since those runtimes won't parse template tags, we
+  // fallback to an equivalent API that assumes no calls
+  // are template tag calls.
+  //
+  // Clients that need to work on older Node runtimes
+  // should not use any part of this API except
+  // calledAsTemplateTagQuick unless that function has
+  // returned true.
+
+  module.exports = function (staticStrings) {
+    // This might be reached if client code is transpiled down to
+    // ES5 but this module is not.
+    throw new Error('ES6 features not supported');
+  };
+  /**
+   * @param {*} firstArg
+   * @param {number} nArgs
+   *
+   * @return {boolean} always false in ES<6 compatibility mode.
+   */
+  module.exports.calledAsTemplateTagQuick = function (firstArg, nArgs) {
+    return false;
+  };
+
+  function stringWrapper() {
+    function TypedString(content) {
+      if (!(this instanceof TypedString)) {
+        return new TypedString(content);
+      }
+      this.content = String(content);
+    }
+    TypedString.prototype.toString = function () {
+      return this.content;
+    };
+    return TypedString;
+  }
+
+  /**
+   * @param {string} content
+   * @constructor
+   */
+  module.exports.Identifier = stringWrapper();
+  /**
+   * @param {string} content
+   * @constructor
+   */
+  module.exports.Fragment = stringWrapper();
```

Comment:
`Identifier` and `Fragment` look like they are identical. Was this an accident or can they just be combined into one thing?

Reply:
And Fragment sounds like it should be using our existing Fragment API: the toSqlString contract. If that is not possible, I need to understand why and then you should remove the toSqlString API completely in favor of the API because there is no reason to have two different Fragment APIs in the same module, that is silly like the module cannot coordinate with itself :)

Reply:
Refactored to use SqlString.raw and SqlString.identifier.